src/dune_engine/rules.ml

```diff
@@ -3,15 +3,24 @@ module Action_builder = Action_builder0
 module Id = Id.Make ()
 
 module Dir_rules = struct
-  module Alias_spec = struct
+  module Alias_spec : sig
     type item =
       | Deps of unit Action_builder.t
       | Action of Rule.Anonymous_action.t Action_builder.t
 
-    type t = { expansions : (Loc.t * item) Appendable_list.t } [@@unboxed]
+    type t = (Loc.t * item) Appendable_list.t
```

Comment:
All these changes seem unrelated to me. Why are we doing them here? They would make it hard to benchmark just the change to the implementation of the `Appendable_list`.

Reply:
(Also, I'm actually changing `src/dune_engine/rules.ml` to make it faster and that moves the ground underneath, seemingly unnecessarily.)

Reply:
Without this change it's impossible to use `Appendable_list.concat` which reduces allocation.

Reply:
Wait, actually I'm not using `Appendable_list.concat anymore`. Let me see if I can get rid of it.